src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -1425,12 +1427,21 @@ def forward(
 
         # Cross-Attention Text
         queries = self.with_pos_embed(hidden_states, position_embeddings)
+        if text_encoder_attention_mask is not None:
+            dtype = text_encoder_hidden_states.dtype
+
+            text_encoder_attention_mask = text_encoder_attention_mask[:, None, None, :]
+            text_encoder_attention_mask = text_encoder_attention_mask.repeat(
+                1, self.num_attention_heads, self.num_queries, 1
+            )
+            text_encoder_attention_mask = text_encoder_attention_mask.to(dtype=dtype)
+            text_encoder_attention_mask = text_encoder_attention_mask * torch.finfo(dtype).min
```

Comment:
Actually - just realised this is in the `DecoderLayer`. This should be done in the decoder, to avoid recreating the repeated attention mask for every decoder layer.

Reply:
https://github.com/huggingface/transformers/pull/30364/commits/0a520b1f5ba9b57938ffc9d2cf6d38575bbb0b6b